challenge-two/src/agent.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Finding,
+  HandleTransaction,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+} from "forta-agent";
+import { UNISWAP_ROUTER_ABI, SWAP_ROUTER_02 } from "./constants";
+
+export const provideHandleTransaction = (
+  abi: any,
+  swapRouter02Address: string
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+    const lowerCaseSwapRouter02Address = swapRouter02Address.toLowerCase();
+
+    // Filter transaction events by logs that match the swap events and router address
+    const swapLogs = txEvent.filterFunction(abi, lowerCaseSwapRouter02Address);
```

Comment:
Right now, this is checking for activity that goes through the Router. However, the bot would fail to detect `swap`s that do not go through the route. E.g. someone `swap`ping directly on the pool. So the bot needs to be updated so that it register _any_ `swap` occurring across any **valid** UniswapV3 pool. I.e. it should not register `swap`s occurring outside of a valid UniswapV3 pool